src/netInfo.re

```diff
@@ -0,0 +1,83 @@
+type infoType = string;
+
+type infoEffectiveType = string;
+
+type connectionType =
```

Comment:
It might be against the convention in this project, but I think we should use the polymorphic generators here. This projects pre-dates the generators and I think it would make sense to use this for new features

https://bucklescript.github.io/docs/en/generate-converters-accessors.html#convert-between-js-string-enum-and-bs-polymorphic-variant

What do you think @wokalski ?

Reply:
@arnarthor I was thinking about that too, and I believe it could simplify the bindings in some way

Reply:
Definitely, we have a lot of these helper functions that only convert between JS and OCaml types.

Reply:
To be honest I don't think that the maintenance difficulty differs. It's slightly more convenient but bindings libraries like this are fairly static (in opposite to ever changing real world applications).

To give you a real argument compare [v0.34 docs](https://facebook.github.io/react-native/docs/0.34/netinfo.html) with [v0.53-RC docs](https://facebook.github.io/react-native/docs/netinfo.html). There's hardly any difference.

Reply:
No I agree, I was more so thinking about this in a way that it's less noise in the repo with these manual conversion functions. But either way is fine by me. Consistency might be better here since everything else is this way.